To: vim_dev@googlegroups.com
Subject: Patch 7.3.077
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.077
Problem:    When updating crypt of swapfile fails there is no error message.
	    (Carlo Teubner)
Solution:   Add the error message.
Files:	    src/memline.c


*** ../vim-7.3.076/src/memline.c	2010-08-15 21:57:31.000000000 +0200
--- src/memline.c	2010-12-08 12:39:10.000000000 +0100
***************
*** 582,587 ****
--- 582,590 ----
  	    idx = ip->ip_index + 1;	    /* go to next index */
  	    page_count = 1;
  	}
+ 
+ 	if (error > 0)
+ 	    EMSG(_("E843: Error while updating swap file crypt"));
      }
  
      mfp->mf_old_key = NULL;
***************
*** 2384,2390 ****
   * Make a copy of the line if necessary.
   */
  /*
!  * get a pointer to a (read-only copy of a) line
   *
   * On failure an error message is given and IObuff is returned (to avoid
   * having to check for error everywhere).
--- 2387,2393 ----
   * Make a copy of the line if necessary.
   */
  /*
!  * Return a pointer to a (read-only copy of a) line.
   *
   * On failure an error message is given and IObuff is returned (to avoid
   * having to check for error everywhere).
***************
*** 2397,2403 ****
  }
  
  /*
!  * ml_get_pos: get pointer to position 'pos'
   */
      char_u *
  ml_get_pos(pos)
--- 2400,2406 ----
  }
  
  /*
!  * Return pointer to position "pos".
   */
      char_u *
  ml_get_pos(pos)
***************
*** 2407,2413 ****
  }
  
  /*
!  * ml_get_curline: get pointer to cursor line.
   */
      char_u *
  ml_get_curline()
--- 2410,2416 ----
  }
  
  /*
!  * Return pointer to cursor line.
   */
      char_u *
  ml_get_curline()
***************
*** 2416,2422 ****
  }
  
  /*
!  * ml_get_cursor: get pointer to cursor position
   */
      char_u *
  ml_get_cursor()
--- 2419,2425 ----
  }
  
  /*
!  * Return pointer to cursor position.
   */
      char_u *
  ml_get_cursor()
***************
*** 2426,2432 ****
  }
  
  /*
!  * get a pointer to a line in a specific buffer
   *
   * "will_change": if TRUE mark the buffer dirty (chars in the line will be
   * changed)
--- 2429,2435 ----
  }
  
  /*
!  * Return a pointer to a line in a specific buffer
   *
   * "will_change": if TRUE mark the buffer dirty (chars in the line will be
   * changed)
*** ../vim-7.3.076/src/version.c	2010-12-08 13:11:15.000000000 +0100
--- src/version.c	2010-12-08 13:15:44.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     77,
  /**/

-- 
An operatingsystem is just a name you give to the rest of bloating
idiosyncratic machine-based-features you left out of your editor.
			(author unknown)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
